### PR TITLE
fix: Always set the API URL for legacy CLI invocations

### PR DESCRIPTION
**Pull request by PeterSchafer**
## Pull Request Submission Checklist
- [x] Follows [CONTRIBUTING](https://github.com/snyk/cli/blob/main/CONTRIBUTING.md) guidelines
- [x] Includes detailed description of changes
- [x] Contains risk assessment (Low | Medium | High)
- [ ] Highlights breaking API changes (if applicable)
- [x] Links to automated tests covering new functionality
- [ ] Includes manual testing instructions (if necessary)
- [ ] Updates relevant GitBook documentation (PR link: ___)
- [ ] Includes product update to be announced in the next stable release notes

## What does this PR do?
The PR ensures to always specify the API URL when running the legacy Typescript CLI. Before the API URL was only set when the API was explicitly set but given the possibility to derive an URL from a OAuth claim, we need to always set the URL.

There was no closed box test for this, which is now there and one test had to be adapted since a side-effect of the change is, that an error message changed. Since the error message is only a side effect of one command and testing the edge case of an incorrectly specified `endpoint`, I removed these unrelated checks from the test.

The Risk of the change should be low, as it only ensures that both golang and typescript CLI actually use the same API URL, making the golang CLI as the single source of truth.

## Where should the reviewer start?
`cliv2/internal/cliv2/cliv2.go`

## How should this be manually tested?
Run the test and observe that despite the fact that no API URL is configured the `SNYK_API` is set to `https://api.my.special.url.io `

```
npx jest test/jest/acceptance/api-auto-detect.spec.ts
```

### CODE REVIEW

File: test/jest/acceptance/api-auto-detect.spec.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+describe('Auto detect API URL', () => {
+  it('Based on OAuth token claim', async () => {
+    const expectedApiUrl = 'https://api.my.special.url.io';
+    const oauthToken = {
+      access_token:
+        'a.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyLCJhdWQiOlsiaHR0cHM6Ly9hcGkubXkuc3BlY2lhbC51cmwuaW8iXX0.b',
+      token_type: 'Bearer',
+      refresh_token: 'configRefreshToken',
+      expiry: '3023-03-29T17:47:13.714448+02:00',
+    };
```
Comment (dotkas) [inline test/jest/acceptance/api-auto-detect.spec.ts:14]: What do you think about something like this for test readability (didn't test it, just for suggestions)

```suggestion
    const payload = {
      sub: '1234567890',
      name: 'John Doe',
      iat: 1516239022,
      aud: [expectedApiUrl],
    };

    const oauthToken = {
      access_token:
        `a.${Buffer.from(
      JSON.stringify(readableTokenPayload)
    ).toString('base64')}.b`,
      token_type: 'Bearer',
      refresh_token: 'configRefreshToken',
      expiry: '3023-03-29T17:47:13.714448+02:00',
    };
```

**Review by dotkas — APPROVED** [review]

File: test/jest/acceptance/api-auto-detect.spec.ts (hunk 1 above)
Comment (PeterSchafer) [reply test/jest/acceptance/api-auto-detect.spec.ts]: Awesome idea! Thank you, just needed some small changes to make it work 😄
